From 3d86a4b909f8e551f36e6a61bce07eb0a10d68b3 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Tue, 18 Feb 2020 13:52:55 +0200
Subject: [PATCH] [SW-8229] goya2: close jtag debug if

Completely close jtag interface by changing pad select of pin from
jtag functionality. Also disable access to dap ROM tables.

Change-Id: Ifeb2c508bd80b93b6f620416feef607520b78661
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/security.c b/common/security.c
index 95a9d96..692258d 100644
--- a/common/security.c
+++ b/common/security.c
@@ -64,10 +64,31 @@
 		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
 }
 
+static void security_disable_jtag_pad_sel(void)
+{
+	/* disable TCK functionality */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_49);
+
+	/* disable TMOD functionality */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_50);
+
+	/* disable DATA IN functionality */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_51);
+
+	/* disable TRSTn functionality */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_52);
+
+	/* disable DATA OUT functionality */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_53);
+}
+
 void security_disable_debug_if(void)
 {
+	regmap_writel(0, mmPSOC_CS_TRACE_TOP_ROM_VALID_0);
+	regmap_writel(0, mmPSOC_CS_TRACE_TOP_ROM_VALID_1);
 	security_disable_i2c_s_mstr_if();
 	security_disable_smi_mstr_if();
+	security_disable_jtag_pad_sel();
 	security_disable_jtag_mstr_if();
 }
 
